### PR TITLE
Feat/inference

### PR DESCRIPTION
**Pull request by yonycherkos**

### CODE REVIEW

File: components/Inference/Inference.cpp [hunk 1]
```diff
@@ -1,70 +1,179 @@
-#include "Inference.h"
-
 
 namespace components{
 
+#include "Inference.h"
+
+using namespace components;
```
Comment (arisema) [inline components/Inference/Inference.cpp]: Already in said namespace.

File: components/Inference/Inference.h [hunk 2]
```diff
@@ -4,26 +4,33 @@
 #include <map>
 #include <set>
 #include <vector>
-#include "KB.h"
```
Comment (arisema) [inline components/Inference/Inference.h:7]: Keep this.

File: components/Inference/Inference.h [hunk 3]
```diff
@@ -4,26 +4,33 @@
...
+#include <tuple>
+#include "../KnowledgeBase/KB.h"
```
Comment (arisema) [inline components/Inference/Inference.h:8]: Remove this.

File: components/Inference/Inference.cpp (hunk 1 above)
Comment (yonycherkos) [reply components/Inference/Inference.cpp]: ok i fixed it

File: components/Inference/Inference.h [hunk 4]
```diff
@@ -4,26 +4,33 @@
...
+  /*
+  *
+  * This class used to infer possible action to be performed by the agent.
+  * and it also updates the knowledgebase with the infered parameters
+  *
+  */
+  private:
+    std::vector<std::pair<int, int>> adjacent_rooms;
+    std::pair<int, int> wumpus_room;
+    components::KnowledgeBase current_kb;
```
Comment (arisema) [inline components/Inference/Inference.h:25]: remove components scope operator. Already in same namespace.

File: components/KnowledgeBase/KB.h [hunk 5]
```diff
@@ -14,22 +14,21 @@ class KnowledgeBase
...
     */
     private:
-        std::map<std::pair<int, int>, DataStructures::Knowledge> data; 
-
-        std::map<std::pair<int, int>, DataStructures::Knowledge> get_data();
+        std::map<std::pair<int, int>, DataStructures::Knowledge> data;
     public:
         KnowledgeBase();
         ~KnowledgeBase();
+        std::map<std::pair<int, int>, DataStructures::Knowledge> get_data();
```
Comment (arisema) [inline components/KnowledgeBase/KB.h:28]: Keep this in private. Unless valid reason.

File: components/KnowledgeBase/KB.h [hunk 6]
```diff
@@ -14,22 +14,21 @@ class KnowledgeBase
...
     * A room is represented by a pair<int, int> as row, col representation.
     * The information collected about each room is held in the value of the map using the Knowledge struct.
     * The knowledge struct (written above) contains all the possible perceptions that can be attributed to a room by the agent after perceptions within the wumpus world.
-    * 
+    *
     */
     private:
-        std::map<std::pair<int, int>, DataStructures::Knowledge> data; 
-
-        std::map<std::pair<int, int>, DataStructures::Knowledge> get_data();
```
Comment (arisema) [inline components/KnowledgeBase/KB.h:26]: Keep this. remove from public. Unless valid reason

File: utilities/DataStructures.h [hunk 7]
```diff
@@ -29,12 +30,17 @@ namespace DataStructures{
         bool scream = false;
         bool wumpus = false;
     };
-    
+
     enum class Decision{
```
Comment (arisema) [inline utilities/DataStructures.h:34]: You can remove this definition.

File: components/KnowledgeBase/KB.h (hunk 5 above)
Comment (yonycherkos) [reply components/KnowledgeBase/KB.h:28]: i will need to access get_data() from inference class.

File: utilities/DataStructures.h [hunk 8]
```diff
@@ -29,12 +30,17 @@ namespace DataStructures{
...
         move_left,
         move_right,
         move_up
     };
-    
+
+    struct Response{
```
Comment (arisema) [inline utilities/DataStructures.h:40]: Please rename to Decision.

File: components/KnowledgeBase/KB.h (hunk 5 above)
Comment (arisema) [reply components/KnowledgeBase/KB.h:28]: Is there a scenario where the access modifiers won't allow you to do that?

**Pull request closed without merging**